core/positions/market_position.go

```diff
@@ -230,3 +230,20 @@ func (p MarketPosition) OrderReducesExposure(ord *types.Order) bool {
 	}
 	return false
 }
+
+// OrderReducesOnlyExposure returns true if the order reduce the position and the extra size if it was to flip the position side.
+func (p MarketPosition) OrderReducesOnlyExposure(ord *types.Order) (reduce bool, extraSize uint64) {
+	// if already closed, or increasing position, we shortcut
+	if p.Size() == 0 || (p.Size() < 0 && ord.Side == types.SideSell) || (p.Size() > 0 && ord.Side == types.SideBuy) {
```

Comment:
Comparison operators and logical AND are with higher precedence than logical OR, but thats okay as well.

Reply:
No, in case for better readability makes sense :+1: